pkgs/test/lib/src/runner/executable_settings.dart

```diff
@@ -45,7 +48,15 @@ class ExecutableSettings {
       if (envVariable != null) return envVariable;
     }
 
-    if (Platform.isMacOS) return _macOSExecutable!;
+    if (Platform.isMacOS) {
+      if (_macOSExectuables case final paths?) {
+        for (final path in paths) {
+          if (p.basename(path) == path) return path;
+          if (p.isAbsolute(path) && File(path).existsSync()) return path;
+        }
+      }
+      throw ArgumentError('Missing macOsExecutable');
```

Comment:
Can we just rely on private field promotion? It is much easier to read imo if so:

```suggestion
      if (_macOSExectuables != null) {
        for (final path in _macOSExectuables) {
          if (p.basename(path) == path) return path;
          if (p.isAbsolute(path) && File(path).existsSync()) return path;
        }
      }
      throw ArgumentError('Missing macOsExecutable');
```